common/src/main/java/com/wynntils/screens/characterselector/widgets/ChangeWorldButton.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.characterselector.widgets;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.screens.base.widgets.WynntilsButton;
+import com.wynntils.screens.characterselector.CharacterSelectorScreen;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.mc.TooltipUtils;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.Texture;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.screens.inventory.tooltip.ClientTooltipComponent;
+import net.minecraft.network.chat.Component;
+
+public class ChangeWorldButton extends WynntilsButton {
+
+    private static final String LOBBY_SERVER = "lobby.wynncraft.com";
+
+    private static final List<Component> TOOLTIP = List.of(
+            Component.translatable("screens.wynntils.characterSelection.changeWorldButton.changeWorld")
+                    .withStyle(ChatFormatting.GREEN),
+            Component.translatable("screens.wynntils.characterSelection.changeWorldButton.description")
+                    .withStyle(ChatFormatting.GRAY));
+    private final CharacterSelectorScreen characterSelectorScreen;
+
+    public ChangeWorldButton(int x, int y, int width, int height, CharacterSelectorScreen characterSelectorScreen) {
+        super(x, y, width, height, Component.literal("Change World"));
+        this.characterSelectorScreen = characterSelectorScreen;
+    }
+
+    @Override
+    public void onPress() {
+        McUtils.sendCommand("hub");
+    }
+
+    @Override
+    public void renderWidget(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        RenderUtils.drawTexturedRect(
+                poseStack,
+                Texture.CHANGE_WORLD_BUTTON.resource(),
+                this.getX(),
+                this.getY(),
+                0,
+                this.width,
+                this.height,
+                0,
+                0,
+                Texture.CHANGE_WORLD_BUTTON.width(),
+                Texture.CHANGE_WORLD_BUTTON.height() / 2,
+                Texture.CHANGE_WORLD_BUTTON.width(),
+                Texture.CHANGE_WORLD_BUTTON.height());
+
+        if (isHovered) {
```

Comment:
Can we pull this if clause out to some method? It's also used in DisconnectButton and PlayButton

Reply:
Yeah, maybe we could

Reply:
Maybe like `private void renderTooltip(PoseStack poseStack, int mouseX, int mouseY, List<Component> tooltip) {
    poseStack.pushPose();
    List<ClientTooltipComponent> clientTooltipComponents =
            TooltipUtils.componentToClientTooltipComponent(tooltip);
    poseStack.translate(
            mouseX
                    - TooltipUtils.getToolTipWidth(
                            clientTooltipComponents, FontRenderer.getInstance().getFont()),
            mouseY - TooltipUtils.getToolTipHeight(clientTooltipComponents),
            100);
    RenderUtils.drawTooltip(poseStack, tooltip, FontRenderer.getInstance().getFont(), true);
    poseStack.popPose();
}`

Reply:
I know the concern is code duplication but sharing this code would mean adding a superclass which might not end up cleaner than copying this once.

Reply:
But I can also see that basically the whole class is duplicated... Up to @DonkeyBlaster if they want it refactored or if it is fine this way.